src/include/transaction/deferred_action_manager.h

```diff
@@ -62,17 +63,24 @@ class DeferredActionManager {
    * Clear the queue and apply as many actions as possible
    * @return numbers of deferred actions processed
    */
-  uint32_t Process(transaction::timestamp_t oldest_txn) {
+  uint32_t Process() {
+    // TODO(John, Ling): this is now more conservative than it needs and can artificially delay garbage collection.
+    //  We should be able to query the cached oldest transaction (should be cheap) in between each event
+    //  and more aggressively clear the backlog abd the deferred event queue
+    timestamp_manager_->CheckOutTimestamp();
```

Comment:
It should be worth a comment why we need to checkout a timestamp here. At least this is not immediately obvious to me.

Reply:
I think we still need this comment